src/main/java/com/compass/vinyl/player/LocalFileSystemRecordPlayer.java

```diff
@@ -162,7 +162,7 @@ public void deleteByTags(List<String> tags, RecordingConfig config) {
                     .forEach(it -> {
                         if (it.toString().endsWith(META_SUFFIX)) {
                             ScenarioMetadata metadata = retrieveMetadataForFile(it, config.getSerializer());
-                            if (metadata.getTags() != null) {
+                            if (metadata != null && metadata.getTags() != null) {
```

Comment:
@mujeebul are we sure, the problem is for null metadata only ? Before why was this not causing an issue? Vinyl is in Production for Android since May, this is coming from the past release only.

Reply:
The issue happens when deserialization returns a null.  hence metadata will also be null.
This could be because of some corrupted cache files.

Reply:
Ok lets merge and verify if this is working.